models/repo_language_stats.go

```diff
@@ -144,3 +144,32 @@ func (repo *Repository) UpdateLanguageStats(commitID string, stats map[string]fl
 
 	return sess.Commit()
 }
+
+// CopyLanguageStat Copy originalRepo language stat information to destRepo (use for forked repo)
+func CopyLanguageStat(originalRepo, destRepo *Repository) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	RepoLang := make(LanguageStatList, 0, 6)
+	if err := sess.Where("`repo_id` = ?", originalRepo.ID).Desc("`percentage`").Find(&RepoLang); err != nil {
+		return err
+	}
+	if len(RepoLang) > 0 {
+		for i := range RepoLang {
+			RepoLang[i].ID = 0
+			RepoLang[i].RepoID = destRepo.ID
+			RepoLang[i].CreatedUnix = timeutil.TimeStampNow()
+		}
+		if _, err := sess.Insert(&RepoLang); err != nil {
+			return err
+		}
+		//update destRepo's indexer status
```

Comment:
can you update the indexer after the loop? since the CommitID should be the same for all RepoLangItems

Reply:
beside this LGTM

Reply:
> can you update the indexer after the loop **once**? since the CommitID should be the same for all RepoLangItems

Thanks. For your advice i think it is unnecessary, the function "updateIndexerStatus"  need to be executed only once--after all the language stat information was written to database. You can view the code above.(like func "UpdateLanguageStats")

Reply:
@L0veSunshine looking forward to your commit ;)

Reply:
> @L0veSunshine looking forward to your commit ;)

I finally understand your means 😂. i had changed.

Reply:
The commitID is the branch commitID, for every language stat item, commit is the same. Don't need to update sereval times.

Reply:
no ... but this is also right

Reply:
Please tell me how should i do.

Reply:
@L0veSunshine -> https://github.com/L0veSunshine/gitea/pull/1

I created a pull to your branch ...

Reply:
this code comment is to long to look at the diff nicely ....

Reply:
@L0veSunshine if you "resolve" this conversation - other mainteiner wont have trublle to look at the diff :)